Fully featured crawler

someshkar opened the pull request:
Uses regex to respect the ```robots.txt``` file and uses a random interval between 1 and 5 seconds to ensure DoS protection services don't kick in.

someshkar committed:
Finish crawler

someshkar committed:
Apply pep8 formatting

someshkar committed:
Add without-bs4.py

someshkar committed:
Added filter for mailto: and tel: links

someshkar committed:
Finish parser

someshkar committed:
Finally fix parser

someshkar committed:
Finish fully featured crawler

someshkar committed:
Revert "Finish fully featured crawler"

This reverts commit 9c3abaa4c8eaa0e3cfcb548621150a25e5bbfe70.

palash-taneja commented:
@someshkar add a gitignore to your thing and exclude the env directory. 
https://github.com/github/gitignore/blob/master/Python.gitignore
Use that

palash-taneja commented:
Otherwise, I think the code is pretty legit :+1: 
If you want more bonus points then I'd recommend keeping a track of visited domains and adding the sleep only if the same domain has been hit > DOMAIN_THRESHOLD times

*Be careful of the distinction b/w domain and webpages*

someshkar committed:
add .gitignore

someshkar committed:
Add domain hit threshold

someshkar commented:
Alright, I've added that ```.gitignore``` file and implemented the domain hit threshold. Also, now the different parts of the code are split into different files for easy manageability as I finish more tasks.

palash-taneja commented:
@someshkar couple of things: 
Do git rm -r env and commit
Also, your crawler fails on my website http://palashtaneja.com even though other crawlers get the URLs on it. 
Traceback (most recent call last):
  File "main.py", line 8, in <module>
    title, links_and_data, current_time, main_source = c.crawl()
  File "/Users/palashtaneja/Github/OSMarathon/open-source-marathon/basic/crawler.py", line 45, in crawl
    domain = str(re.findall(r'://(.*?)/', link)[0])
IndexError: list index out of range

someshkar committed:
remove env

someshkar commented:
I removed the env, as for the error, @sayamkanwar was running into a similar issue on his machine for whatever reason with my basic parser, while it worked flawlessly on my machine.

someshkar committed:
Fix Palash's error

someshkar commented:
Turns out this error started showing up on my machine too, and that was because my regex wouldn't match the root domain (https://techcrunch.com) and would throw an index out of range error. I've fixed it now. Here's an asciinema recording of the output on my machine:

https://asciinema.org/a/KhwE2epiqJgJBWCuUBhQfqdQt

someshkar committed:
Uncomment domain hit threshold

palash-taneja commented:
Good job, approving :+1:

palash-taneja reviewed (approved):
Works
